docs/source/concepts/event.rst

```diff
@@ -9,8 +9,133 @@ The Event System
    :local:
    :backlinks: none
 
-Outline
--------
- - Event creators
- - Event listeners
- - Perils of an event system in a simulation.
+
+``vivarium`` constructs and manages the flow of :ref:`time <time_concept>`
+through the emission of regularly scheduled events. This event system provides
+a means of coordinating across various components in a simulation.
+
+What is an Event?
+-----------------
+
+An :class:`Event <vivarium.framework.event.Event>` is a container for a series
+of attributes that provide all the necessary information to respond to the event,
+including an index into the population table that contains all affected
+simulants, the simulation time at which the event was emitted, the time step
+size at the time of emission, and any additional user-provided data. ``vivarium``
+manages these events by requiring that each type of event be unique
+(e.g., there is only one ``simulation_end`` event and its type (or name - the
+two are equivalent because of the uniqueness) is "simulation_end"). Individual
+components are then allowed to register ``listeners`` (i.e., callable functions
+to respond on the emission of an event) to each event. When an event is emitted,
+all listeners that have registered themselves to that event type are called.
+
+Lifecycle Events
+----------------
+
+The simulation engine itself is responsible for the emission of a dedicated set
+of events that determine the progression of time in the simulation. Each time
+step in the simulation corresponds to the emission of a set of ``time_step``
+events. :ref:`Components <components_concept>` can register themselves as
+listeners to these events and thus take action on each time step or other
+key simulation phase.
+
+The following table depicts the events emitted by the simulation engine itself
+and the lifecycle phase when they are emitted. See the
+:ref:`lifecycle concept note <lifecycle_concept>` for more information about
+these phases.
+
+.. list-table:: **Events Emitted by the Simulation Engine**
+    :header-rows: 1
+    :widths: 30, 40
+
+    *   - Lifecycle Phase
+        - Events
+    *   - | Post-setup
+        - | ``post_setup``
+    *   - | Main Event Loop
+        - | ``time_step_prepare``
+          | ``time_step``
+          | ``time_step__cleanup``
+          | ``collect_metrics``
+    *   - | Finalization
+        - | ``simulation_end``
+
+
+Interacting with Events
+-----------------------
+
+The :class:`EventInterface <vivarium.framework.event.EventInterface>` is available
+off the :ref:`Builder <builder_concept>` and provides two options for interacting
+with the event system:
+
+1. :func:`register_listener <EventInterface.register_listener()>` to add a
+listener to a given event to be called on emission
+
+2. :func:`get_emitter <vivarium.framework.event.EventInterface.get_emitter()>` to
+retrieve a callable emitter for a given event
+
+Although methods for both getting emitters and registering listeners are provided,
```

Comment:
This is  a good candidate for a .. note::  directive to make it stick out more

Reply:
I put a longer .. caution:: directive below in the section about emitting events explaining that you shouldn't actually do it and why not but if you think it would be helpful to have it highlighted here too, I can definitely add that

Reply:
The only consideration I have is that people will stop reading when they get the info they want, so attaching the warning close to the information is helpful